### PR TITLE
fix: don't able to call server when promotional code is empty

### LINKED ISSUE #3815: Promotional Code empty still calls server
**Describe the bug**
The Apply promo code section while ordering tickets for an event hits the API even when the promocode text field is empty.

**To Reproduce**
1. Open the page for ordering ticket
2. Click on apply promocode
3. Click on tick without entering any value -Hits the API

**Expected behaviour**
Shouldn't hit api and should be handled client side.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #3815 

#### Short description of what this resolves:
don't able to call server when promotional code is empty. So making the submit button disabled when promotional code is empty and also added validation to the field.
full flow of the promotional code field : https://youtu.be/34cZDbxJDDU

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by maze-runnar** [comment]
@kushthedude @codesun please review.

File: app/templates/components/public/ticket-list.hbs [hunk 1]
```diff
@@ -125,9 +125,10 @@
                       placeholder=(t 'Promotional Code')
                       readonly=promotionalCodeApplied
                       key-up=(action 'handleKeyPress')
+                      id='promotionalCode'
               }}
               {{#unless promotionalCodeApplied}}
-                <div role="button" class="ui icon button" {{action 'applyPromotionalCode'}}>
+                <div role="button" class="ui icon {{if promotionalCode '' 'disabled'}} button" {{action 'applyPromotionalCode'}}>
```
Comment (kushthedude) [inline app/templates/components/public/ticket-list.hbs]: @maze-runnar Is this really needed ? Validation should be taking care for empty input .

Comment (maze-runnar) [reply app/templates/components/public/ticket-list.hbs]: but it was still calling the server and reloading the page.

Comment (kushthedude) [reply app/templates/components/public/ticket-list.hbs]: Are you sure ?

**Pull request merged**